dbms/src/Interpreters/InterpreterFactory.cpp

```diff
@@ -145,6 +147,10 @@ std::unique_ptr<IInterpreter> InterpreterFactory::get(ASTPtr & query, Context &
     {
         return std::make_unique<InterpreterDescribeQuery>(query, context);
     }
+    else if (typeid_cast<ASTExplainQuery *>(query.get()))
+    {
+        return std::make_unique<InterpreterExplainQuery>(query, context);
```

Comment:
We can add a setting to enable this query,
because API is not stabilized for general usage.
... and because this is not the EXPLAIN query that our users want.

(they want something like query plan with annotations)

Reply:
enable_debug_queries